packages/protocol/contracts/L1/libs/LibProvingAlt.sol

```diff
@@ -109,7 +109,7 @@ library LibProvingAlt {
 
         // The new proof must meet or exceed the minimum tier required by the
         // block or the previous proof; it cannot be on a lower tier.
-        if (proof.tier == 0 || proof.tier < meta.minTier || proof.tier < ts.tier) {
+        if (proof.tier == 0 || proof.tier < ts.tier) {
```

Comment:
add the optimistic name check here.

Reply:
Dani, I reviewed the code more carefully. I think we should simply remove:

```
  if (ts.tier < meta.minTier) {
                revert L1_UNEXPECTED_TRANSITION_TIER();
            }
```

and keep `if (proof.tier == 0 || proof.tier < meta.minTier || proof.tier < ts.tier)`.


The reason we have questions now, is because previously 
```
  if (ts.tier < meta.minTier) {
                revert L1_UNEXPECTED_TRANSITION_TIER();
            }
```
is 

```
  assert (ts.tier >= meta.minTier);
```
this is to really make sure we know assert must be true. That's why we use asset, but once asserts are converted to if-revert, it will cause a confusion that the case may actually happen.


So maybe we should **keep other asserts there without converting them to if-revert.**

Reply:
Okay, so just keep as is, and remove what you suggests here. But what about assert ? (Shall we put this back:
`assert (ts.tier >= meta.minTier);` ? I guess no)